### ISSUE TITLE
Windows: Cannot complete update to 1.66.4 or uninstall, due to version mismatch

### ISSUE THREAD

**Issue by zhayes0887** [tailscale/tailscale#12232]
### What is the issue?

On Windows, I am unable to update to 1.66.4 from 1.64.0 due to Windows is listing the current installed version as 1.62.1. Where as, the application and the website are listing 1.64.0 is installed on this machine. The process runs fine until it attempts to "Remove Old Versions", it then pops up stating that "The Feature you are trying to use is on a network resource that is not available. On this pop up, it states that it is looking for 'tailscale-setup-1.62.1-amd64.msi'. I also cannot uninstall the application in its current state. There is no way around this issue.

![App_AboutTailscale_Version](https://github.com/tailscale/tailscale/assets/24834973/2a22b696-dcdc-48aa-be5f-2f10a559b023)
![WindowsListing_WrongVersion](https://github.com/tailscale/tailscale/assets/24834973/3dea3159-5644-48b3-94f4-bcbb803242b0)
![NetworkResourceUnavail_LookingForOldInstaller](https://github.com/tailscale/tailscale/assets/24834973/e84891af-2699-4bba-ac8c-2fd1960e822f)
![FileCannotBeFound_OldInstallerVersionListed](https://github.com/tailscale/tailscale/assets/24834973/65984a4a-e96e-4576-9cec-51b502c54855)

### Steps to reproduce

1. Install Tailscale Version 1.62.1 on Windows
2. Update to Version 1.64.0
3. Attempt to update to Version 1.66.4, or uninstall Tailscale.
4. Observe the user is unable to update or uninstall due to a version mismatch and the application looking for an old installer package.

### OS

Windows

### OS version

Windows 10 Pro Version 10.0.19045 Build 19045

### Tailscale version

1.64.0

**Comment by dblohm7** [comment]
@zhayes0887 that error is indicating that Windows cannot find the cached MSI package for the older release. Have you run CCleaner or some other tool that might be trying to free up hard disk space by deleting cached MSI packages?

**Comment by zhayes0887** [comment]
No, I do not use tools like CCleaner, because it can blindly delete files like that. However, I do clean out my downloads folder regularly and will not keep old MSIs for the sake of uninstall later. If the uninstall process requires the user to keep the MSI for that version there should be a copy that is kept in the program's working directory and not left up to the use to hold on to them. Especially in a location like Downloads.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by dblohm7** [comment]
@zhayes0887 the user is not required to keep anything; Windows caches a copy in a special directory under the Windows directory. That's why I asked about CCleaner.

Are you running any Antivirus other than Windows Defender? We've had a history of AV software falsely flagging our binaries and installer packages as malware and quarantining them.

**Comment by zhayes0887** [comment]
Nothing more than defender, also nothing flagged there. I just followed the normal updates process until it started failing. Then I noticed the version mismatch between what Windows was reporting (1.62.1), what was listed in the app (1.64.0), and the version in the MSI's filename(1.62.1). The machines page for Tailscale was also reporting 1.64.0.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by dblohm7** [comment]
@zhayes0887 I'm not sure how your machine arrived in that state, but to fix it, you can easily re-obtain the MSI packages for the older versions and provide them as prompted; hopefully that will be enough to make Windows happy.

Just go to https://pkgs.tailscale.com/stable/ , go to the `View older version` dropdown and select the version you need, and download it.

**Comment by croakerbcts** [comment]
@zhayes0887 can I confirm if the above resolved your issue?

**Comment by zhayes0887** [comment]
I ended up having to manually remove the files/ect to uninstall, the version it was stating I needed and it was not available. But Yes after I got that install removed it started to work again.
